pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -906,3 +915,50 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+type DockerswarmSDConfig struct {
+	// Address of the docker daemon
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Host string `json:"host"`
+	// ProxyConfig allows customizing the proxy behaviour for this scrape config.
+	// +optional
+	v1.ProxyConfig `json:",inline"`
```

Comment:
(to check) I don't think that we need comment/markers for inline fields.
```suggestion
	v1.ProxyConfig `json:",inline"`
```